### PR TITLE
Add formatter and linter

### PR DESCRIPTION
**Pull request by AndrewKorzh**
# Title

Add formatter and linter

# Description

In this PR, a formatter and linter has been added

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -169,6 +169,15 @@ and will notify you about number of found captchas on the page.
...
 
+## For Developers
+Before each push or PR, run in the root directory of the project:
+
+```bash
+ruff check
+
+ruff format 
+```
+
```
Comment (MatthewZMSU) [inline README.md:180]: Move it to `CONTIBUTING.md`

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+[tool.ruff]
+line-length = 88
+fix = true
+
+[tool.ruff.lint]
+select = ["F", "C", "W"]
```
Comment (MatthewZMSU) [inline pyproject.toml:6]: Don't we want to check order of imports too?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[tool.ruff]
+line-length = 88
+fix = true
```
Comment (MatthewZMSU) [inline pyproject.toml:3]: With GitHubActions we only find problems. No need to fix them inplace, it doesn't make any sense cause the repository WON'T be fixed with this approach.

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+[tool.ruff.lint]
+select = ["F", "C", "W"]
+extend-ignore = ["E203", "E501", "F401", "C408", "F811", "N"]
+
+[tool.ruff.format]
```
Comment (MatthewZMSU) [inline pyproject.toml:9]: add `indent-width`

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+indent-style = "space"
+line-ending = "lf"
+quote-style = "double"
+skip-magic-trailing-comma = true
```
Comment (MatthewZMSU) [inline pyproject.toml:13]: trailing comma must be in the code due to Back code style.
Delete this

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+line-length = 88
+fix = true
+
+[tool.ruff.lint]
+select = ["F", "C", "W"]
+extend-ignore = ["E203", "E501", "F401", "C408", "F811", "N"]
+
+[tool.ruff.format]
+indent-style = "space"
+line-ending = "lf"
```
Comment (MatthewZMSU) [inline pyproject.toml:11]: Well, we want scrapy-puppeteer to be cross-platform. Maybe we should set it to `native`.
@yatskov, what do you think?

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+[tool.ruff]
+line-length = 88
```
Comment (MatthewZMSU) [inline pyproject.toml:2]: Well, this is the default value. If you want it to be so, leave it as is and resolve the conversation.

File: pyproject.toml (hunk 4 above)
Comment (MatthewZMSU) [inline pyproject.toml:9]: Think about `docstring-code-format`.

File: pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+fix = true
+
+[tool.ruff.lint]
+select = ["F", "C", "W"]
+extend-ignore = ["E203", "E501", "F401", "C408", "F811", "N"]
```
Comment (MatthewZMSU) [inline pyproject.toml:7]: `extend-ignore` is deprecated now: https://docs.astral.sh/ruff/settings/#lint_extend-ignore. Try to use something different.

Comment (MatthewZMSU) [inline pyproject.toml:7]: I think we should not ignore `N`. Maybe we are allowed to ignore `N807` and even select `N`.
Here you may do this as you wish.

File: .github/workflows/python-test.yml [hunk 9]
```diff
@@ -29,6 +30,18 @@ jobs:
...
+
+      - name: Format Code with Ruff
+        run: |
+          ruff format .
+
+      - name: Run Ruff Check
+        run: |
+          ruff check .
+
       - name: Run Tests
```
Comment (MatthewZMSU) [inline .github/workflows/python-test.yml:45]: Do not use linter and formatter in `tests`. Put them into a separate job section. I prefer installing `ruff` via `pipx`.

**Review by MatthewZMSU — CHANGES_REQUESTED** [review]

**Comment by MatthewZMSU** [comment]
You better be cautious with `pyproject.toml` files. Poetry also uses it, and some packages can be installed with it. Think about using `ruff.toml` but it is not necessary in our case.

File: README.md (hunk 1 above)
Comment (AndrewKorzh) [reply README.md:180]: Done

File: pyproject.toml (hunk 2 above)
Comment (AndrewKorzh) [reply pyproject.toml:6]: Added

File: pyproject.toml (hunk 3 above)
Comment (AndrewKorzh) [reply pyproject.toml:3]: now it is false)

File: pyproject.toml (hunk 4 above)
Comment (AndrewKorzh) [reply pyproject.toml:9]: Done

File: pyproject.toml (hunk 5 above)
Comment (AndrewKorzh) [reply pyproject.toml:13]: done

File: pyproject.toml (hunk 6 above)
Comment (AndrewKorzh) [reply pyproject.toml:11]: now it is "native"

**Pull request closed without merging**